sdk/servicebus/service-bus/src/core/messageReceiver.ts

```diff
@@ -952,9 +942,25 @@ export class MessageReceiver extends LinkEntity {
           connectionHost: this._context.namespace.config.host,
           delayInSeconds: 15
         };
-        if (!this.wasCloseInitiated) {
-          await retry<void>(config);
-        }
+        await retry<void>(config);
+      } else if (causedByDisconnect) {
```

Comment:
This and the following else is for the case when we have encountered a non retryable error. Can we do an early exit for this? Helps with readability. 



```
if (translatedError && !translatedError.retryable) {
    // Log:

Reply:
I've updated so that this and the following else occur first and exit early. I'm not sure much is gained but I did add more comments around why each is needed/what each does.